eo-runtime/src/test/eo/org/eolang/float-tests.eo

```diff
@@ -255,71 +272,157 @@
       0.2
     $.equal-to 155.0
 
-[] > zero-div-zero-is-equal-to-nan
+[] > zero-div-zero-as-bytes-is-equal-to-nan-as-bytes
   assert-that > @
     (0.0.div 0.0).as-bytes
     $.equal-to nan.as-bytes
 
-[] > float-actions-with-nan-are-nan
-  42.5 > num!
-  nan > cached-nan!
-  cached-nan.as-bytes > nan-as-bytes!
+[] > float-times-nan-is-nan
+  nan > not-a-number!
   assert-that > @
-    and.
-      (num.times cached-nan).as-bytes.eq nan-as-bytes
-      (num.plus cached-nan).as-bytes.eq nan-as-bytes
-      (num.minus cached-nan).as-bytes.eq nan-as-bytes
-      (num.div cached-nan).as-bytes.eq nan-as-bytes
-    $.equal-to TRUE
+    as-bytes.
+      35.2.times not-a-number
+    $.equal-to not-a-number.as-bytes
 
-[] > float-comparings-with-nan-are-false
-  42.5 > num!
-  nan > cached-nan!
+[] > float-plus-nan-is-nan
+  nan > not-a-number!
+  assert-that > @
+    as-bytes.
+      35.2.plus not-a-number
+    $.equal-to not-a-number.as-bytes
 
+[] > float-minus-nan-is-nan
+  nan > not-a-number!
   assert-that > @
-    and.
-      (num.eq cached-nan).eq FALSE
-      (num.lt cached-nan).eq FALSE
-      (num.lte cached-nan).eq FALSE
-      (num.gt cached-nan).eq FALSE
-      (num.gte cached-nan).eq FALSE
-    $.equal-to TRUE
+    as-bytes.
+      35.2.minus not-a-number
+    $.equal-to not-a-number.as-bytes
```

Comment:
@maxonfjvipon If `as-bytes` makes problems in the pipelines, why are you using it here? is it different `as-bytes`?

Reply:
@volodya-lombrozo see above about NaN checking